src/Sql/dbo/Stored Procedures/Organization_EnableCollectionEnhancements.sql

```diff
@@ -0,0 +1,299 @@
+CREATE PROCEDURE [dbo].[Organization_EnableCollectionEnhancements]
+    @OrganizationId UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    -- Step 1: AccessAll migration for Groups
+        -- Create a temporary table to store the groups with AccessAll = 1
+        SELECT [Id] AS [GroupId], [OrganizationId]
+        INTO #TempGroup
+        FROM [dbo].[Group]
+        WHERE [AccessAll] = 1
+          AND [OrganizationId] = @OrganizationId;
+
+        -- Create a temporary table to store distinct OrganizationUserIds
+        SELECT DISTINCT GU.[OrganizationUserId]
+        INTO #TempOrganizationUsers
+        FROM [dbo].[GroupUser] GU
+            JOIN #TempGroup TG ON GU.[GroupId] = TG.[GroupId];
+
+        BEGIN TRY
+            BEGIN TRANSACTION;
+                -- Update existing rows in [dbo].[CollectionGroup]
+                UPDATE CG
+                SET
+                    CG.[ReadOnly] = 0,
+                    CG.[HidePasswords] = 0,
+                    CG.[Manage] = 0
+                FROM [dbo].[CollectionGroup] CG
+                INNER JOIN [dbo].[Collection] C ON CG.[CollectionId] = C.[Id]
+                INNER JOIN #TempGroup TG ON CG.[GroupId] = TG.[GroupId]
+                WHERE C.[OrganizationId] = TG.[OrganizationId];
+
+                -- Insert new rows into [dbo].[CollectionGroup]
+                INSERT INTO [dbo].[CollectionGroup] ([CollectionId], [GroupId], [ReadOnly], [HidePasswords], [Manage])
+                SELECT C.[Id], TG.[GroupId], 0, 0, 0
+                FROM [dbo].[Collection] C
+                INNER JOIN #TempGroup TG ON C.[OrganizationId] = TG.[OrganizationId]
+                LEFT JOIN [dbo].[CollectionGroup] CG ON CG.[CollectionId] = C.[Id] AND CG.[GroupId] = TG.[GroupId]
+                WHERE CG.[CollectionId] IS NULL;
+
+                -- Update Group to clear AccessAll flag
+                UPDATE G
+                SET [AccessAll] = 0
+                FROM [dbo].[Group] G
+                INNER JOIN #TempGroup TG ON G.[Id] = TG.[GroupId]
+
+                -- Execute User_BumpAccountRevisionDateByOrganizationUserId for each unique OrganizationUserId
+                DECLARE @Step1OrganizationUserId UNIQUEIDENTIFIER
+
+                DECLARE UniqueOrgUserIdCursor CURSOR FOR
+                SELECT [OrganizationUserId]
+                FROM #TempOrganizationUsers
+
+                OPEN UniqueOrgUserIdCursor
+                FETCH NEXT FROM UniqueOrgUserIdCursor INTO @Step1OrganizationUserId
+
+                WHILE (@@FETCH_STATUS = 0)
+                BEGIN
+                    -- Execute the stored procedure for the current OrganizationUserId
+                    EXEC [dbo].[User_BumpAccountRevisionDateByOrganizationUserId] @Step1OrganizationUserId
```

Comment:
There is a bulk version of this sprocs which will accept multiple orgUser Ids. If that's cleaner